apps/heft/src/utilities/subprocess/SubprocessRunnerBase.ts

```diff
@@ -157,13 +268,106 @@ export abstract class SubprocessRunnerBase<TSubprocessConfiguration> {
     }
 
     if (willUseInspector) {
-      // Don't bother instantiating a Terminal object just for this one logging statement.
-      terminalProvider.write(
-        `Subprocess with inspector bound to port ${inspectPort}${terminalProvider.eolCharacter}`,
-        TerminalProviderSeverity.log
-      );
+      terminal.writeLine(`Subprocess with inspector bound to port ${inspectPort}`);
     }
 
     return nodeArgs;
   }
+
+  private _receiveMessageFromParentProcess(message: ISubprocessMessageBase): void {
+    for (const subprocessCommunicationManager of this._subprocessCommunicationManagers) {
+      if (subprocessCommunicationManager.canHandleMessageFromParentProcess(message)) {
+        subprocessCommunicationManager.receiveMessageFromParentProcess(message);
+        return;
+      }
+    }
+
+    throw new Error(
+      'Subprocess communication manager. No communication manager can handle message type ' +
+        `"${message.type}" from parent process.`
+    );
+  }
+
+  private _receiveMessageFromSubprocess(message: ISubprocessMessageBase): void {
+    for (const subprocessCommunicationManager of this._subprocessCommunicationManagers) {
+      if (subprocessCommunicationManager.canHandleMessageFromSubprocess(message)) {
+        subprocessCommunicationManager.receiveMessageFromSubprocess(message);
+        return;
+      }
+    }
+
+    throw new Error(
+      'Subprocess communication manager. No communication manager can handle message type ' +
+        `"${message.type}" from subprocess.`
+    );
+  }
+
+  public static serializeArg(arg: unknown): ISubprocessApiCallArg {
```

Comment:
Could you add some unit tests for serializing/deserializing?  Some Jest snapshots of the protocol would be very enlightening.